src/main/java/com/taboola/backstage/Backstage.java

```diff
@@ -1,13 +1,13 @@
 package com.taboola.backstage;
 
-import com.taboola.backstage.internal.*;
+import com.taboola.backstage.internal.*
 import com.taboola.backstage.internal.config.CommunicationConfig;
 import com.taboola.backstage.internal.config.SerializationConfig;
 import com.taboola.backstage.internal.factories.BackstageEndpointsFactory;
 import com.taboola.backstage.internal.factories.BackstageEndpointsRetrofitFactory;
-import com.taboola.backstage.internal.BackstageInternalTools;
-import com.taboola.backstage.internal.BackstageInternalToolsImpl;
-import com.taboola.backstage.services.*;
+import com.taboola.backstage.services.*
+import java.util.List;
+import okhttp3.Interceptor;
```

Comment:
@amitharris lets impl same interface for interceptor and wrap by okhttp3 inside the code.
We dont want to expose specific connection pool we using for future ability of changing it without breaking backward computability